### ISSUE TITLE
Where to debug next?

### ISSUE THREAD

**Issue by Rodeoclash** [refile/refile#289]
Rails 4.2

Generating paths in my models like:
```
  def profile_image_thumbnail_path
    Refile.attachment_url(self, :profile_image, :fill, 300, 300, format: 'jpg')
  end
```

And serialising to the frontend, however:

```
/attachments/store/fill/300/300/b558bc9415e4e53c9fa5d32fdf9162b5bcecc2c2c96c2895fbb5a189c861/profile_image.jpg
```

Returns a 404 (not found). Logged as: ```Refile::App: Could not find attachment by id```

However in the Rails console

```
f = Freelancer.first
f.profile_image_id
b558bc9415e4e53c9fa5d32fdf9162b5bcecc2c2c96c2895fbb5a189c861
f.profile_image.exists?
=> true
```

No problems with finding it directly. Also, everything works great in my local development, I only get the 404s in production. I can see files being generated by the correct ID in the file system. I can also read files directly into the console as binary, so unlikely permissions.

Bit stuck on where to debug next, suggestion most welcome. Confirmed that imagemagick is installed with ```identify``` command.

**Comment by Rodeoclash** [comment]
Ok, so accessing the console in dokku will actually instantiate a new docker container instance which goes away when you exit out of again. Apologies for posting the 'please help me' style post.

**Comment by BlitzKraft** [comment]
I am facing the same issue. How did you resolve it? 

This happened when accessed the rails console while the rails server is running.

**Comment by Rodeoclash** [comment]
@BlitzKraft My problem was essentially a misunderstanding about how dokku worked, i.e. console access did not attach to a running container and instead created its own.

Are you running docker/docker-compose/dokku?

**Comment by BlitzKraft** [comment]
No I'm not running any of them. 

I had a different issue. When I made the changes, I also changed a few lines in my products.rb (models) file. That was what's causing the images to not upload and throwing 404s for exisiting images. It created a somewhat similar problem and google led me to your comments. 

I corrected my model and everything is working fine now. Well, not fine, but as expected (I'm still in dev mode).
